pkg/monitor/cluster/clustersync.go

```diff
@@ -0,0 +1,73 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (mon *Monitor) emitClusterSync(ctx context.Context) error {
+	if mon.hiveClusterManager == nil {
+		// TODO(hive): remove this once we have Hive everywhere.
+		mon.log.Info("skipping: no hive cluster manager")
+		return nil
+	}
+
+	clusterSync, err := mon.hiveClusterManager.GetClusterSync(ctx, mon.doc)
+	if err != nil {
+		return err
+	}
+	if clusterSync != nil {
+		if clusterSync.Status.SyncSets != nil {
```

Comment:
Similar comment here. I think it makes more sense to get rid of some of the nesting and exit early if clusterSync.Status.SelectorSyncSets is nil.
```
if clusterSync.Status.SyncSets == nil {
     return nil 
}

Reply:
suggested changes implemented